src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -112,7 +112,7 @@ trait Types
   /** The current skolemization level, needed for the algorithms
    *  in isSameType, isSubType that do constraint solving under a prefix.
    */
-  private val _skolemizationLevel = Parallel.Counter()
+  private val _skolemizationLevel = Parallel.WorkerThreadLocal(0)
```

Comment:
I see you changed your mind ^^
Good, in `SynchronizedTypes` it's `TheredLocal` as well.